### PR TITLE
feat: add pnpm support for installing CLI

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
Enhancement, feature

Refs https://github.com/webpack/webpack-cli/pull/2040#issuecomment-723059281

**Did you add tests for your changes?**
No

**Does this PR introduce a breaking change?**
No

**What needs to be documented once your changes are merged?**
Nothing

### CODE REVIEW

File: bin/webpack.js [hunk 1]
```diff
@@ -83,8 +83,18 @@ if (!cli.installed) {
 	console.error(notify);
 
 	const isYarn = fs.existsSync(path.resolve(process.cwd(), "yarn.lock"));
+	const isPnpm = fs.existsSync(path.resolve(process.cwd(), "pnpm-lock.yaml"));
```
Comment (evilebottnawi) [inline bin/webpack.js:86]: Let's avoid multiple fs call, it reduce performance, let's do it:
```js
const isNpm = fs.existsSync(path.resolve(process.cwd(), "package-lock.json"));

let isYarn = false;

if (!isNpm) {
  isYarn = fs.existsSync(path.resolve(process.cwd(), "yarn.lock"));
}

let isPnpm = false;

if (!isYarn) { 
  isPnpm = fs.existsSync(path.resolve(process.cwd(), "pnpm-lock.yaml"))
}

```

npm is most popular, then `yarn`, then `pnpm`, so we avoid extra fs calls for `npm`

Comment (snitin315) [reply bin/webpack.js]: Let's improve it in `webpack-cli` too.

File: bin/webpack.js [hunk 2]
```diff
@@ -82,9 +82,30 @@ if (!cli.installed) {
 
 	console.error(notify);
 
-	const isYarn = fs.existsSync(path.resolve(process.cwd(), "yarn.lock"));
+	const isNpm = fs.existsSync(path.resolve(process.cwd(), "package-lock.json"));
```
Comment (jamesgeorge007) [inline bin/webpack.js:85]:
```suggestion
	const isNpm = fs.existsSync(path.resolve("package-lock.json"));
```

File: bin/webpack.js [hunk 3]
```diff
@@ -82,9 +82,30 @@ if (!cli.installed) {
...
+
+	let isYarn = false;
+
+	if (!isNpm) {
+		isYarn = fs.existsSync(path.resolve(process.cwd(), "yarn.lock"));
```
Comment (jamesgeorge007) [inline bin/webpack.js:90]:
```suggestion
		isYarn = fs.existsSync(path.resolve("yarn.lock"));
```

File: bin/webpack.js [hunk 4]
```diff
@@ -82,9 +82,30 @@ if (!cli.installed) {
...
+	}
+
+	let isPnpm = false;
+
+	if (!isYarn) {
+		isPnpm = fs.existsSync(path.resolve(process.cwd(), "pnpm-lock.yaml"));
```
Comment (jamesgeorge007) [inline bin/webpack.js:96]:
```suggestion
		isPnpm = fs.existsSync(path.resolve("pnpm-lock.yaml"));
```

File: bin/webpack.js (hunk 2 above)
Comment (snitin315) [reply bin/webpack.js:85]: Maybe this is required here too?

**Review by evilebottnawi — APPROVED** [review]

**Review by rishabh3112 — APPROVED** [review]

**Review by sokra — CHANGES_REQUESTED** [review]

**Review by sokra — APPROVED** [review]

**Comment by sokra** [comment]
Thanks

**Pull request merged**